types/line-navigator/index.d.ts

```diff
@@ -0,0 +1,56 @@
+// Type definitions for line-navigator 2.1
+// Project: https://github.com/anpur/client-line-navigator
+// Definitions by: Vladimir Poluch <https://github.com/vlapo>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace LineNavigator {
+    interface Options {
+        /**
+         * Encoding
+         * @default 'utf8'
+         */
+        encoding?: string;
+        /**
+         * Size of chunk
+         * @default 4096
+         */
+        chunkSize?: number;
+        /**
+         * Return error when line is longer than chunkSize, otherwise it will be threated as several lines.
+         * @default false
+         */
+        throwOnLongLines?: boolean;
+    }
+
+    interface FindMatch {
+        line: string;
+        offset: number;
+        length: number;
+    }
+
+    interface FindAllResult extends FindMatch {
+        index: string;
+    }
+
+    type ReadLinesCallback = (err: any, index: number, lines: string[] | undefined, isEof: boolean | undefined, progress: number | undefined) => void;
+    type FindCallback = (err: any, index: number | undefined, match: FindMatch | undefined) => void;
```

Comment:
the same, just err can be undefined accordion to documentation

Reply:
Yes it is but in case `err` is filled arguments`index` and `match` are undefined (according to code).